fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/EliminateEmptyRelation.java

```diff
@@ -109,15 +109,21 @@ public List<Rule> buildRules() {
                     if (union.getConstantExprsList().isEmpty()) {
                         Plan child = nonEmptyChildren.get(0);
                         List<Slot> unionOutput = union.getOutput();
-                        List<Slot> childOutput = child.getOutput();
-                        List<NamedExpression> projects = Lists.newArrayList();
-                        for (int i = 0; i < unionOutput.size(); i++) {
-                            ExprId id = unionOutput.get(i).getExprId();
-                            Alias alias = new Alias(id, childOutput.get(i), unionOutput.get(i).getName());
-                            projects.add(alias);
-                        }
+                        int childIdx = union.children().indexOf(nonEmptyChildren.get(0));
+                        if (childIdx >= 0) {
```

Comment:
UT is a better way to test it, because regression test could not ensure child's output is different with regularChildOutput